lab/src/raft/leader_election.go

```diff
@@ -1,11 +1,32 @@
 package raft
 
+import "time"
+
 func (rf *Raft) SetHeatBeatClock() {
 	rf.heartBeatClock.Reset(rf.clockInterval*time.Millisecond)
 }
 
 func (rf *Raft) SendHeartbeat() {
-	// send heartbeat to all other followers
+	// Send heartbeat to all other followers
+	for server := 0; server < len(rf.peers); server++ {
+		if server == rf.me {
+			continue
+		}
+		rf.mu.Lock()
+		currentTerm := rf.currentTerm
+		rf.mu.Unlock()
+		go func(currentTerm int, server int) {
+			args := AppendEntriesArgs{currentTerm, rf.me, []*LogEntry{}, 0, 0, rf.committedIndex}
+			reply := AppendEntriesReply{}
+			ok := rf.SendAppendEntries(server, &args, &reply)
+			if !ok {
+				return
+			}
+			if !reply.Success && reply.Term > currentTerm {
```

Comment:
We only have to check `reply.Succes` here.

Reply:
Done